File: AINoval/lib/models/setting_type.dart
// AINoval/lib/models/setting_type.dart
enum SettingType {
  character('CHARACTER', '角色'),
  location('LOCATION', '地点'),
  item('ITEM', '物品'),
  lore('LORE', '背景知识'),
  faction('FACTION', '组织/势力'),
  event('EVENT', '事件'),
  concept('CONCEPT', '概念/规则'),
  creature('CREATURE', '生物/种族'),
  magicSystem('MAGIC_SYSTEM', '魔法体系'),
  technology('TECHNOLOGY', '科技设定'),
  culture('CULTURE', '文化'),
  history('HISTORY', '历史'),
  organization('ORGANIZATION', '组织'),
  // —— 通用叙事/世界构建扩展 ——
  worldview('WORLDVIEW', '世界观'),
  pleasurePoint('PLEASURE_POINT', '爽点'),
  anticipationHook('ANTICIPATION_HOOK', '期待感钩子'),
  theme('THEME', '主题'),
  tone('TONE', '基调'),
  style('STYLE', '文风'),
  trope('TROPE', '母题/套路'),
  plotDevice('PLOT_DEVICE', '剧情装置'),
  powerSystem('POWER_SYSTEM', '力量体系'),
  goldenFinger('GOLDEN_FINGER', '金手指'),
  timeline('TIMELINE', '时间线'),
  religion('RELIGION', '宗教'),
  politics('POLITICS', '政治'),
  economy('ECONOMY', '经济'),
  geography('GEOGRAPHY', '地理'),
  other('OTHER', '其他');

  const SettingType(this.value, this.displayName);
  final String value;
  final String displayName;

  // 为了向后兼容，添加 key 属性
  String get key => value;

  static SettingType fromValue(String value) {
    return SettingType.values.firstWhere(
      (e) => e.value == value.toUpperCase(),
      orElse: () => SettingType.other,
    );
  }

  // 添加 JSON 序列化支持
  static SettingType fromJson(dynamic json) {
    if (json is String) {
      return fromValue(json);
    } else if (json is Map<String, dynamic>) {
      return fromValue(json['value'] ?? json['key'] ?? 'OTHER');
    }
    return SettingType.other;
  }

  Map<String, dynamic> toJson() => {
    'value': value,
    'displayName': displayName,
  };
}

// Helper for UI if needed
class SettingTypeOption {
  final SettingType type;
  bool isSelected;

  SettingTypeOption(this.type, {this.isSelected = false});
} 

File: AINovalServer/src/main/java/com/ainovel/server/domain/model/SettingType.java
package com.ainovel.server.domain.model;

import com.fasterxml.jackson.annotation.JsonValue;

/**
 * 小说设定类型枚举
 * 定义了可以由AI生成或用户创建的设定类型。
 * 与前端 AINoval/lib/models/setting_type.dart (或类似路径下的常量/枚举) 保持一致。
 */
public enum SettingType {
    CHARACTER("CHARACTER", "角色"),
    LOCATION("LOCATION", "地点"),
    ITEM("ITEM", "物品"),
    LORE("LORE", "背景知识"), // 世界观/传说/规则等
    FACTION("FACTION", "组织/势力"),
    EVENT("EVENT", "事件"),
    CONCEPT("CONCEPT", "概念/规则"), // 细化的规则或抽象概念
    CREATURE("CREATURE", "生物/种族"),
    MAGIC_SYSTEM("MAGIC_SYSTEM", "魔法体系"),
    TECHNOLOGY("TECHNOLOGY", "科技设定"),
    CULTURE("CULTURE", "文化"),
    HISTORY("HISTORY", "历史"),
    ORGANIZATION("ORGANIZATION", "组织"),
    // —— 通用叙事/世界构建扩展 ——
    WORLDVIEW("WORLDVIEW", "世界观"),
    PLEASURE_POINT("PLEASURE_POINT", "爽点"),
    ANTICIPATION_HOOK("ANTICIPATION_HOOK", "期待感钩子"),
    THEME("THEME", "主题"),
    TONE("TONE", "基调"),
    STYLE("STYLE", "文风"),
    TROPE("TROPE", "母题/套路"),
    PLOT_DEVICE("PLOT_DEVICE", "剧情装置"),
    POWER_SYSTEM("POWER_SYSTEM", "力量体系"),
    GOLDEN_FINGER("GOLDEN_FINGER", "金手指"),
    TIMELINE("TIMELINE", "时间线"),
    RELIGION("RELIGION", "宗教"),
    POLITICS("POLITICS", "政治"),
    ECONOMY("ECONOMY", "经济"),
    GEOGRAPHY("GEOGRAPHY", "地理"),
    OTHER("OTHER", "其他");

    private final String value;
    private final String displayName;

    SettingType(String value, String displayName) {
        this.value = value;
        this.displayName = displayName;
    }

    @JsonValue // This annotation is important for Jackson serialization/deserialization
    public String getValue() {
        return value;
    }

    public String getDisplayName() {
        return displayName;
    }

    public static SettingType fromValue(String value) {
        for (SettingType type : values()) {
            if (type.value.equalsIgnoreCase(value)) {
                return type;
            }
        }
        // Fallback to OTHER or throw an exception if strict matching is required
        // throw new IllegalArgumentException("Unknown setting type: " + value); 
        return OTHER; 
    }

    @Override
    public String toString() {
        return this.value;
    }
} 